src/libraries/System.Runtime.Numerics/src/System/Number.BigInteger.cs

```diff
@@ -466,17 +466,28 @@ internal static BigInteger ParseBigInteger(ReadOnlySpan<char> value, NumberStyle
             if (value.IsEmpty)
             {
                 // There's nothing beyond significant leading block. Return it as the result.
-                if ((int)(leading ^ signBits) < 0 || (int)leading == int.MinValue)
+                if ((int)(leading ^ signBits) < 0)
                 {
-                    // The sign of result differs with leading digit, or it's int.MinValue.
+                    // The sign of result differs with leading digit.
                     // Require to store in _bits.
-                    result = new BigInteger((int)signBits | 1, [leading]);
+                    if ((int)signBits < 0)
+                    {
+                        // Negative value
+                        result = leading == 0
+                            ? new BigInteger(-1, [leading, 1u])
+                            : new BigInteger(-1, [unchecked((uint)-(int)leading)]);
+                    }
+                    else
+                    {
+                        // Positive value
+                        result = new BigInteger(1, [leading]);
+                    }
                     return ParsingStatus.OK;
                 }
                 else
                 {
-                    // Small value that fits in _sign.
-                    result = new BigInteger((int)leading, null);
+                    // Small value that fits in Int32.
+                    result = new BigInteger((int)leading);
```

Comment:
I removed the `(int)leading == int.MinValue` condition and changed it to call `new BigInteger(int)`.
It avoids allocation by using `BigInteger.s_bnMinInt`.